src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3438 @@
+
+    if (yaw < 0x3000) {
+        SubS_SetOfferMode(&this->stateFlags, SUBS_OFFER_MODE_ONSCREEN, SUBS_OFFER_MODE_MASK);
+    } else {
+        SubS_SetOfferMode(&this->stateFlags, SUBS_OFFER_MODE_NONE, SUBS_OFFER_MODE_MASK);
+    }
+
+    this->stateFlags &= ~ENAN_STATE_ENGAGED;
+    return 1;
+}
+
+/* End of HandleShedule functions */
+
+void EnAn_HandleSchedule(EnAn* this, PlayState* play) {
+    switch (this->scheduleResult) {
+        case ANJU_SCH_RECEIVE_LETTER_FROM_POSTMAN:
+        case ANJU_SCH_ATTEND_GORON:
+        case ANJU_SCH_GIVE_LUNCH_TO_GRANNY:
+            EnAn_HandleSch_InteractActor(this, play);
+            break;
+
+        case ANJU_SCH_LAUNDRY_POOL_SIT:
+            EnAn_HandleSch_LaundryPool(this, play);
+            break;
+
+        case ANJU_SCH_WAITING_FOR_KAFEI:
+            EnAn_HandleSch_WaitingForKafei(this, play);
+            break;
+
+        case ANJU_SCH_WITH_KAFEI:
+            EnAn_HandleSch_WithKafei(this, play);
+            break;
+
+        case ANJU_SCH_RANCH:
+        case ANJU_SCH_COOKING:
+        case ANJU_SCH_RECEPTIONIST_IDLE:
+        case ANJU_SCH_WAITING_CLOSING_TIME:
+        case ANJU_SCH_SWEEPING:
+        case ANJU_SCH_TALKING_WITH_MOM:
+            EnAn_HandleSch_80B572D4(this, play);
+            break;
+
+        case ANJU_SCH_MIDNIGHT_MEETING:
+            EnAn_HandleSch_80B572D4(this, play);
+            break;
+
+        case ANJU_SCH_DOOR_26:
+        case ANJU_SCH_DOOR_27:
+        case ANJU_SCH_DOOR_28:
+        case ANJU_SCH_DOOR_29:
+        case ANJU_SCH_DOOR_30:
+        case ANJU_SCH_DOOR_31:
+        case ANJU_SCH_DOOR_32:
+        case ANJU_SCH_DOOR_33:
+        case ANJU_SCH_DOOR_34:
+        case ANJU_SCH_DOOR_35:
+        case ANJU_SCH_DOOR_36:
+        case ANJU_SCH_DOOR_37:
+        case ANJU_SCH_DOOR_38:
+        case ANJU_SCH_DOOR_39:
+            EnAn_HandleSch_Door(this, play);
+            break;
+
+        case ANJU_SCH_WALKING_40:
+        case ANJU_SCH_WALKING_42:
+        case ANJU_SCH_WALKING_43:
+        case ANJU_SCH_WALKING_44:
+        case ANJU_SCH_WALKING_45:
+        case ANJU_SCH_WALKING_46:
+        case ANJU_SCH_WALKING_47:
+        case ANJU_SCH_WALKING_48:
+        case ANJU_SCH_WALKING_49:
+        case ANJU_SCH_WALKING_50:
+        case ANJU_SCH_WALKING_51:
+        case ANJU_SCH_WALKING_52:
+        case ANJU_SCH_WALKING_53:
+        case ANJU_SCH_WALKING_54:
+        case ANJU_SCH_WALKING_55:
+        case ANJU_SCH_WALKING_56:
+        case ANJU_SCH_WALKING_57:
+        case ANJU_SCH_WALKING_58:
+        case ANJU_SCH_WALKING_59:
+        case ANJU_SCH_WALKING_60:
+        case ANJU_SCH_WALKING_61:
+        case ANJU_SCH_WALKING_62:
+        case ANJU_SCH_WALKING_63:
+            EnAn_HandleSch_FollowTimePath(this, play);
+            break;
+
+        default:
+            break;
+    }
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 3, 0x2AA8);
+}
+
+void EnAn_Initialize(EnAn* this, PlayState* play) {
```

Comment:
Hmm, this function name is awkward, because it is really close to `EnAn_Init`. I called a similar function in EnFall `EnFall_Setup`, though that might be confused with functions that setup action functions...what do you think? It's hard to think of a good name for this that isn't already "reserved" in some way.

Reply:
Yeah, it is a hard one to name.
`Setup` would be bad because this function is an actionfunc, so it could introduce even more confusion.

An alternative is to call it `WaitForObject`, but itself doesn't wait for objects to be loaded, other function does... so idk

Reply:
Perhaps something like `FinishInit/InitFinish`?

Reply:
I guess I'll do `FinishInit`